File: ios/Classes/apiOnly/CardValidation.swift
import Adyen

class CardValidation {
    func validateCardNumber(cardNumber: String, enableLuhnCheck: Bool) -> Bool {
        CardNumberValidator(isLuhnCheckEnabled: enableLuhnCheck, isEnteredBrandSupported: true).isValid(cardNumber)
    }
    
    func validateCardExpiryDate(expiryMonth: String, expiryYear: String) -> Bool {
        let lastTwoYearChars = String(expiryYear.suffix(2))
        return CardExpiryDateValidator().isValid("\(expiryMonth)\(lastTwoYearChars)")
    }
    
    func validateCardSecurityCode(securityCode: String, cardBrand: String?) -> Bool {
        guard let cardBrand else {
            return CardSecurityCodeValidator().isValid(securityCode)
        }
        let cardType = CardType(rawValue: cardBrand)
        return CardSecurityCodeValidator(cardType: cardType).isValid(securityCode)
    }
}

File: android/src/main/kotlin/com/adyen/checkout/flutter/apiOnly/CardValidation.kt
package com.adyen.checkout.flutter.apiOnly

import com.adyen.checkout.core.common.CardBrand
import com.adyen.checkout.core.common.helper.CardExpiryDateValidationResult
import com.adyen.checkout.core.common.helper.CardExpiryDateValidator
import com.adyen.checkout.core.common.helper.CardNumberValidationResult
import com.adyen.checkout.core.common.helper.CardNumberValidator
import com.adyen.checkout.core.common.helper.CardSecurityCodeValidationResult
import com.adyen.checkout.core.common.helper.CardSecurityCodeValidator

internal object CardValidation {
    fun validateCardNumber(cardNumber: String, enableLuhnCheck: Boolean): Boolean =
        CardNumberValidator.validateCardNumber(cardNumber, enableLuhnCheck) is CardNumberValidationResult.Valid

    fun validateCardExpiryDate(expiryMonth: String, expiryYear: String): Boolean {
        if (!expiryMonth.matches(Regex("\\d{2}")) || !expiryYear.matches(Regex("\\d{2}"))) {
            return false
        }
        return CardExpiryDateValidator.validateExpiryDate(expiryMonth, expiryYear) is CardExpiryDateValidationResult.Valid
    }

    fun validateCardSecurityCode(securityCode: String, cardBrand: String?): Boolean {
        val result = CardSecurityCodeValidator.validateSecurityCode(
            securityCode = securityCode,
            cardBrand = cardBrand?.let(::CardBrand),
        )
        return result is CardSecurityCodeValidationResult.Valid
    }
}
